py3status/modules/speedtest_cli.py

```diff
@@ -0,0 +1,301 @@
+    {server_url} server url, eg 'http://speedtest.telecom-paristech.fr/upload.php'
+    {upload} upload rate to speedtest server, human readable
+    {upload_raw} upload rate to speedtest server, for format comparation
+    {upload_unit} unit used for upload, eg 'MB/s'
+
+Color thresholds:
+    format:
+        xxx: print a color based on the value of `xxx` placeholder
+
+The module will be triggered on clic only. Not at start.
+
+Note that all placeholders have a version prefixed by `previous_`, eg `previous_download`.
+This can be usefull to compare two runs.
+
+Requires:
+    speedtest-cli: Command line interface for testing internet bandwidth using speedtest.net
+
+Examples:
+```
+# colored based on thresholds
+speedtest_cli {
+    thresholds = {
+        "download": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+        "ping": [(200, "bad"), (150, "orange"), (100, "degraded"), (10, "good")],
+        "quality": [
+            ("ok", "good"),
+            ("bad", "degraded"),
+            ("faster", "good"),
+            ("slower", "degraded"),
+            ("faster", "good"),
+        ],
+        "upload": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+    }
+    format = '[\?color=ping {ping} ms] [\?color=upload ↑ {upload} {upload_unit}]'
+    format += ' [\?color=download ↓ {download} {download_unit}]'
+}
+
+# colored based on comparation between two last runs
+speedtest_cli {
+    format = ' [[\?if=download_raw<previous_download_raw&color=degraded ↓]|[\?color=ok ↑]]'
+}
+
+# compare only download
+speedtest_cli {
+    format = '[[\?if=quality=slower&color=download_raw ↓ {download} {download_unit}]'
+    format += '[\?if=quality=faster&color=download_raw ↑ {download} {download_unit}]'
+    format += '[\?if=quality=ok&color=download_raw -> {download} {download_unit}]'
+    format += '[\?if=quality=bad&color=download_raw x {download} {download_unit}]]'
+    format += '|'
+}
+```
+
+@author Cyril Levis (@cyrinux)
+
+SAMPLE OUTPUT
+{'full_text': u'\u25cf 208.18 ms \u2191 0.247 MB/s \u2193 0.453 MB/s'}
+"""
+
+from json import loads
+
+STRING_NOT_INSTALLED = "not installed"
+MISSING_PLACEHOLDER = "{download} or {upload} placeholder required"
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    button_refresh = 2
+    button_share = None
+    format = (
+        u"\u25cf [{ping} ms] [↑ {upload} {upload_unit}] [↓ {download} {download_unit}]"
+    )
+    server_id = None
+    si_units = False
+    sleep_timeout = 5
+    thresholds = {
+        "download": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+        "ping": [(200, "bad"), (150, "orange"), (100, "degraded"), (10, "good")],
+        "quality": [
+            ("ok", "good"),
+            ("bad", "degraded"),
+            ("faster", "good"),
+            ("slower", "degraded"),
+            ("faster", "good"),
+        ],
+        "upload": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+    }
+    timeout = 10
```

Comment:
`--timeout TIMEOUT     HTTP timeout in seconds. Default 10`.

(Opinion) It's good that we try to make things possible. However, I think we should just kill this option.

Reply:
I have recently use case with poor connection I want to test and where setting timeout help me ...